### ISSUE TITLE
improve behaviour on missing /dev/disk/by-id directory

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2401]
Thanks to forum member Omx for highlighting this failing. In some incompatible hardware configurations, there can be no by-id sub-directory created by udev: such as when no device has a serial number. Currently we fail with an inelegant:

> Houston, we've had a problem:
> "... "/usr/bin/ls: cannot access ‘/dev/disk/by-id’: No such file or directory" ..."

It would be a far better experience to catch this specific error and advise Web-UI users as to there likely being an incompatibility regarding a complete lack of device serial, with a reference to the original no by-id directory error just in case. We could also link to our existing Quick start: Minimum system requirements doc section:
https://rockstor.com/docs/installation/quickstart.html#minimum-system-requirements

LInking for context to the associated forum post:
https://forum.rockstor.com/t/error-running-a-command-cmd-usr-bin-ls-lr-dev-disk-by-id-rc-2-stdout-stderr-usr-bin-ls-cannot-access-dev-disk-by-id-no-such-file-or-directory/8510/2

**Comment by phillxnet** [comment]
```
cmd = /usr/bin/ls -lr /dev/disk/by-id
rc = 2
stdout = ['']
stderr = ["/usr/bin/ls: cannot access '/dev/disk/by-id': No such file or directory", '']
```
And later knock-ons such as:

`[21/Mar/2023 15:15:45] ERROR [smart_manager.data_collector:994] Failed to update disk state.. exception: 500 Server Error: Internal Server Error for url: http://127.0.0.1:8000/api/disks/scan`

**Comment by phillxnet** [comment]
Once a system is in this state, post setup screen, it is possible to by-hand visit the Web-UI via direct URL manipulation. And upon visiting the disk page and using the rescan option the same error is then displayed: representing a post install  reproducer.

# Post proposed fix
We have our existing no-serial message displayed

![no_by-id_dir_on_system](https://user-images.githubusercontent.com/2521585/226693909-b1177053-f94d-418b-9e12-b8b50d4ac74c.png)
rwise functional disk unusable message:

along with the following info log entry:
```
[21/Mar/2023 17:27:58] INFO [system.osi:1876] -- /dev/disk/by-id missing. See 'Minimum system requirements' in docs. --
[21/Mar/2023 17:27:58] INFO [storageadmin.views.disk:152] Deleting duplicate or fake (by serial) disk db entry. Serial = (fake-serial-b9bffdd5-752e-4a8f-b818-38da8f9cc0a2).
```

**Comment by phillxnet** [comment]
Our later code, with it's associated logging, now enabled by catching this earliest instance (via pending fix); also explains it's own failure re no by-id name / no serial: e.g.
```
[21/Mar/2023 17:41:45] INFO [system.osi:1876] -- /dev/disk/by-id missing. See 'Minimum system requirements' in docs. --
[21/Mar/2023 17:41:45] INFO [storageadmin.views.disk:152] Deleting duplicate or fake (by serial) disk db entry. Serial = (fake-serial-55634742-cf51-4781-be36-a6887108da77).
[21/Mar/2023 17:41:45] ERROR [storageadmin.views.command:112] Skipping pool (ROOT) mount as attached disk (vda3) has no by-id name (no serial # ?)
[21/Mar/2023 17:41:46] ERROR [storageadmin.middleware:33] Exception occurred while processing a request. Path: /api/commands/refresh-share-state method: POST
```

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2519
